src/sc_client/constants/sc_type/sc_type.py

```diff
@@ -0,0 +1,198 @@
+
+    def is_const(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_CONST) != 0
+
+    def is_var(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_VAR) != 0
+
+    def is_pos(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_POS_ARC) == bitmasks.SC_TYPE_POS_ARC
+
+    def is_neg(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NEG_ARC) == bitmasks.SC_TYPE_NEG_ARC
+
+    def is_fuz(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_FUZ_ARC) == bitmasks.SC_TYPE_FUZ_ARC
+
+    def is_perm(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_PERM_ARC) == bitmasks.SC_TYPE_PERM_ARC
+
+    def is_temp(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_TEMP_ARC) == bitmasks.SC_TYPE_TEMP_ARC
+
+    def is_actual(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_ACTUAL_ARC) == bitmasks.SC_TYPE_ACTUAL_ARC
+
+    def is_inactual(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_INACTUAL_ARC) == bitmasks.SC_TYPE_INACTUAL_ARC
+
+    def is_tuple(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_TUPLE) == bitmasks.SC_TYPE_NODE_TUPLE
+
+    def is_structure(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_STRUCTURE) == bitmasks.SC_TYPE_NODE_STRUCTURE
+
+    def is_struct(self) -> bool:
+        warnings.warn("ScType `is_struct` method is deprecated. Use `is_structure` method instead.", DeprecationWarning)
+        return self.is_structure()
+
+    def is_role(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_ROLE) == bitmasks.SC_TYPE_NODE_ROLE
+
+    def is_no_role(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_NO_ROLE) == bitmasks.SC_TYPE_NODE_NO_ROLE
+
+    def is_norole(self) -> bool:
+        warnings.warn("ScType `is_norole` method is deprecated. Use `is_no_role` method instead.", DeprecationWarning)
+        return self.is_no_role()
+
+    def is_class(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_CLASS) == bitmasks.SC_TYPE_NODE_CLASS
+
+    def is_superclass(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_SUPERCLASS) == bitmasks.SC_TYPE_NODE_SUPERCLASS
+
+    def is_material(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_MATERIAL) == bitmasks.SC_TYPE_NODE_MATERIAL
+
+    def is_valid(self) -> bool:
+        return self.__bool__()
+
+    def is_equal(self, other: ScType) -> bool:
+        return self.__eq__(other)
+    
+    def _is_not_compatible_by_mask(self, new_type: ScType, mask):
+        subtype = self.value & mask
+        new_subtype = new_type.value & mask
+        return subtype != bitmasks.SC_TYPE_UNKNOWN and subtype != new_subtype
+    
+    def _is_expendable_to(self, new_type: ScType): # it is equal to `sc_storage_is_type_expendable_to` in the sc-machine
+        self_value = self.value
+        new_value = new_type.value
```

Comment:
these two aren't needed here
later they are assigned to a variable of type ScType, so their name is not correct as well

Reply:
There are needed, see below